src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,39 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const { id, title, completed, userId } = selectedTodo;
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(userId)
+      .then(setUser)
+      .finally(() => setIsLoading(false));
+  }, [userId]);
+
+  const handleModalClose = () => {
+    setSelectedTodo(null);
+  };
 
-export const TodoModal: React.FC = () => {
   return (
```

Comment:
You're interacting with the DOM directly by using the 'className' attribute. In React, it's recommended to avoid direct DOM manipulation as much as possible. Instead, you can use the 'classnames' library to define classes conditionally. You can find more information about it [here](https://www.npmjs.com/package/classnames).